### DISCUSSION TITLE
Consider using pre-signed URLs for uploads

### DISCUSSION THREAD

**Discussion by pratik-kanthi** [doshareme/src#22] [Ideas]
Hi there,

I'm uncertain how much this repository reflects the overall scope of the project. Assuming that the uploads will be routed through the server, this is a bottleneck eventually. AWS S3 provides the option to generate a presigned url which can be used by a client to upload directly to S3 avoid the server entirely. A lambda callback can then update the mongoDB record.

Does this align with the project's goals?

**Comment by vednig** [comment]
Yeah, these issues are part of optimizing for performance #21
